src/core/unittest/RecvBufferTest.cpp

```diff
@@ -49,8 +91,33 @@ struct RecvBuffer {
     bool HasUnreadData() {
         return QuicRecvBufferHasUnreadData(&RecvBuf) != FALSE;
     }
+    uint32_t ReadBufferNeededCount() {
+        return QuicRecvBufferReadBufferNeededCount(&RecvBuf);
+    }
+    QUIC_STATUS ProvideChunks(CXPLAT_LIST_ENTRY& Chunks) {
+        auto Result = QuicRecvBufferProvideChunks(&RecvBuf, &Chunks);
+        Dump();
+        return Result;
+    }
+    QUIC_STATUS ProvideChunks(
+        const std::vector<uint32_t>& Sizes,
+        size_t BufferSize,
+        _In_reads_bytes_(BufferSize) uint8_t* Buffer
+        )
+    {
+        CXPLAT_LIST_ENTRY ChunkList;
+        QUIC_STATUS Status = MakeAppOwnedChunks(Sizes, BufferSize, Buffer, &ChunkList);
+        if (Status != QUIC_STATUS_SUCCESS) {
+            return Status;
+        }
+        Status = ProvideChunks(ChunkList);
+        if (Status != QUIC_STATUS_SUCCESS) {
+            FreeChunkList(ChunkList);
+        }
+        return Status;
+    }
     void IncreaseVirtualBufferLength(uint32_t Length) {
-        QuicRecvBufferIncreaseVirtualBufferLength(&RecvBuf, Length);
+        QuicRecvBufferTryIncreaseVirtualBufferLength(&RecvBuf, Length);
     }
```

Comment:
Should this helper also return the boolean? Should we have unittest to verify it's behavior?

Reply:
I guess a unit test won't hurt - not for the helper itself, it is pretty trivial, but we have not validation for the "increase the virtual length then write" scenario.